packages/frontend/src/components/Explorer/CustomMetricWriteBackModal/index.tsx

```diff
@@ -92,9 +94,25 @@ const SingleCustomMetricModalContent = ({
         isLoading,
     } = useWriteBackCustomMetrics(projectUuid!);
     const [showDiff, setShowDiff] = useState(true);
+    const [error, setError] = useState<string | undefined>();
 
     const previewCode = useMemo(() => {
-        return yaml.dump({ [item.name]: convertCustomMetricToDbt(item) });
+        try {
+            const code = yaml.dump({
+                [item.name]: convertCustomMetricToDbt(item),
+            });
+            setError(undefined);
+            return code;
+        } catch (e) {
+            if (e instanceof NotImplementedError) {
```

Comment:
[non-blocker] could it also throw this error? 

https://github.com/lightdash/lightdash/blob/a821783c614cf5f3649705f5ffad93ba454ed467/packages/common/src/types/filterGrammarConversion.ts#L30

Reply:
yes !, it could, as well as other LightdashError , 
If that happens, we will still show the error message on `setError(getErrorMessage(e));`

`NotImplementedError` is just the most "expected" error and what we're trying to solve here, therefore, the error message is more detailed.

That being said, I'm going to make a small change to also show the `error type`if not NotImpementedError

Reply:
Updated: 

![image](https://github.com/user-attachments/assets/2d2bb0c2-2fbd-4ead-a7bd-c87b044c5d6a)